File: yh_sum/src/yh_sum_pub.cpp
#include "ros/ros.h"
#include "yh_sum/yh_sum_msg.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_sum_pub");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_sum::yh_sum_msg>("yh_sum_topic", 100);

    ros::Rate loop_rate(5);

    yh_sum::yh_sum_msg msg;

    int cnt = 0;

    while(ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;
        cnt++;
        pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;

}

File: yh_sum/scripts/yh_sum_pub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_sum.msg import yh_sum_msg

def talker():
   
    pub = rospy.Publisher("yh_sum_topic",yh_sum_msg,queue_size=100)
    rospy.init_node("yh_sum_pub",anonymous=True)
    loop_rate = rospy.Rate(10)
    cnt = 0
    msg = yh_sum_msg()


    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt 
        rospy.loginfo("send message: %d", msg.stamp.secs)
        rospy.loginfo("send message: %d", msg.stamp.nsecs)
        rospy.loginfo("send message: %d", msg.data)

        cnt += 1
        pub.publish(msg)
        loop_rate.sleep() 
        
if __name__=="__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass